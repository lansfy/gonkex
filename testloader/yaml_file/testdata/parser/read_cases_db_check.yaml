- name: loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to dbChecks
  method: GET
  path: /some/path
  response:
    200: "aaabbb"
  dbChecks:
    - dbQuery: "SELECT {{ .foo }}"
      dbResponse:
        - '{{ .bar }} item1'
        - '{{ .bar }} item2'
      comparisonParams:
         ignoreArraysOrdering: true
         ignoreValues: true
    - dbQuery: "UPDATE {{ .foo }}"
      dbResponse:
        - 'item3 {{ .bar }}'
      comparisonParams:
         disallowExtraFields: true
  cases:
    - dbQueryArgs:
        foo: value1
      dbResponseArgs:
        bar: value2
    - dbQueryArgs:
        foo: value21
      dbResponseArgs:
        bar: value22
