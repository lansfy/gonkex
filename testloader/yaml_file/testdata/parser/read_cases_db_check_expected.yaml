- GetName: 'loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to dbChecks #1'
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: "aaabbb"
  GetDatabaseChecks:
    - DbQueryString: "SELECT value1"
      DbResponseJson:
        - value2 item1
        - value2 item2
      GetComparisonParams:
        IgnoreArraysOrdering: true
        IgnoreValuesChecking: true
    - DbQueryString: "UPDATE value1"
      DbResponseJson:
        - item3 value2 
      GetComparisonParams:
        DisallowExtraFields: true
  GetFileName: testdata/parser/read_cases_db_check.yaml
  OneOfCase: true
  FirstTestInFile: true

- GetName: 'loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to dbChecks #2'
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: "aaabbb"
  GetDatabaseChecks:
    - DbQueryString: "SELECT value21"
      DbResponseJson:
        - value22 item1
        - value22 item2
      GetComparisonParams:
        IgnoreArraysOrdering: true
        IgnoreValuesChecking: true
    - DbQueryString: "UPDATE value21"
      DbResponseJson:
        - item3 value22
      GetComparisonParams:
        DisallowExtraFields: true
  GetFileName: testdata/parser/read_cases_db_check.yaml
  OneOfCase: true
  LastTestInFile: true
